src/main/java/org/veupathdb/service/eda/common/client/EdaSubsettingClient.java

```diff
@@ -105,18 +106,21 @@ public ResponseFuture getTabularDataStream(
   public CloseableIterator<Map<String, String>> getTabularDataIterator(String studyId,
                                                                        List<APIFilter> variableFilters,
                                                                        StreamSpec streamSpec) {
+    // Convert everything to internal subsetting classes to bridge the gap from merging/data services.
     final Study study = Resources.getStudyResolver().getStudyById(studyId);
 
     // Use metadata cache directly, which bypasses user studies, since user studies don't currently have files.
     final boolean fileBasedSubsetting = Resources.getMetadataCache().studyHasFiles(study.getStudyId());
 
+    // Resolve schema based on whether study is user or curated.
     final String schemaName = resolveSchema(study);
-    LOG.info("Resolved schema name: {} from study: {} with study source type: {}", schemaName, studyId, study.getStudySourceType());
+    LOG.debug("Resolved schema name: {} from study: {} with study source type: {}", schemaName, studyId, study.getStudySourceType());
+    final List<Filter> internalFilters = ApiConversionUtil.toInternalFilters(study, variableFilters, schemaName);
```

Comment:
why break this out into a separate line, when only used once?

Reply:
I wondered if cramming so much stuff into the method call was a contributor to not noticing the bug.